src/sntp.rs

```diff
@@ -188,3 +200,22 @@ impl Drop for EspSntp {
         info!("Dropped");
     }
 }
+
+pub struct EspSntpSubscription {
+    // FIXME
+    // TBD
+}
+
+#[allow(dead_code)]
+struct Newtype<T>(pub T);
+
+impl<T> From<Newtype<T>> for unsafe extern "C" fn(*mut esp_idf_sys::timeval) {
+    fn from(ntype: Newtype<T>) -> Self {
+        let boxed = Box::new(ntype.0);
+        let ptr: *mut T = Box::into_raw(boxed);
+
+        unsafe {
+            std::mem::transmute::<*mut T, unsafe extern "C" fn(*mut esp_idf_sys::timeval)>(ptr)
```

Comment:
@ivmarkov is this approach acceptable, or is there a simpler way to go about this?

Reply:
Ignore the above, the transmute causes the closure to loose context of its argument - so this is a bad way to approach this.  Open to suggestions, @ivmarkov thanks.